rust/src/tx_builder.rs

```diff
@@ -399,10 +456,117 @@ impl TransactionBuilder {
         };
     }
 
+    /// Set explicit auxiliary data via an AuxiliaryData object
+    /// It might contain some metadata plus native or Plutus scripts
     pub fn set_auxiliary_data(&mut self, auxiliary_data: &AuxiliaryData) {
         self.auxiliary_data = Some(auxiliary_data.clone())
     }
 
+    /// Set metadata using a GeneralTransactionMetadata object
+    /// It will be set to the existing or new auxiliary data in this builder
+    pub fn set_metadata(&mut self, metadata: &GeneralTransactionMetadata) {
+        let mut aux = self.auxiliary_data.as_ref().cloned().unwrap_or(AuxiliaryData::new());
+        aux.set_metadata(metadata);
+        self.set_auxiliary_data(&aux);
+    }
+
+    /// Add a single metadatum using TransactionMetadatumLabel and TransactionMetadatum objects
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_metadatum(&mut self, key: &TransactionMetadatumLabel, val: &TransactionMetadatum) {
+        let mut metadata = self.auxiliary_data.as_ref()
+            .map(|aux| { aux.metadata().as_ref().cloned() })
+            .unwrap_or(None)
+            .unwrap_or(GeneralTransactionMetadata::new());
+        metadata.insert(key, val);
+        self.set_metadata(&metadata);
+    }
+
+    /// Add a single JSON metadatum using a TransactionMetadatumLabel and a String
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_json_metadatum(
+        &mut self,
+        key: &TransactionMetadatumLabel,
+        val: String,
+    ) -> Result<(), JsError> {
+        self.add_json_metadatum_with_schema(key, val, MetadataJsonSchema::NoConversions)
+    }
+
+    /// Add a single JSON metadatum using a TransactionMetadatumLabel, a String, and a MetadataJsonSchema object
+    /// It will be securely added to existing or new metadata in this builder
+    pub fn add_json_metadatum_with_schema(
+        &mut self,
+        key: &TransactionMetadatumLabel,
+        val: String,
+        schema: MetadataJsonSchema,
+    ) -> Result<(), JsError> {
+        let metadatum = encode_json_str_to_metadatum(val, schema)?;
+        self.add_metadatum(key, &metadatum);
+        Ok(())
+    }
+
+    /// Set explicit Mint object to this builder
+    /// it will replace any previously existing mint
+    pub fn set_mint(&mut self, mint: &Mint) {
+        self.mint = Some(mint.clone());
+    }
+
+    /// Add a mint entry to this builder using a PolicyID and MintAssets object
+    /// It will be securely added to existing or new Mint in this builder
```

Comment:
What exactly do we mean by "securely"? I see it used in a lot of these comments.

Reply:
Means that if there already IS `Mint` set to the builder - it will be added to that object, but if there's none yet - new one will be created. Basically means that you are adding new entries without having to bother checking if the map itself been created or not - that is taken care of internally here.

Open for any suggestions for better wording than that.